src/accelerate/test_utils/scripts/test_ops.py

```diff
@@ -94,6 +107,45 @@ def test_reduce_mean(state):
     assert torch.allclose(reduced_tensor, truth_tensor), f"{reduced_tensor} != {truth_tensor}"
 
 
+def test_gather_for_metrics_with_iterable_dataset(state):
+    assert state.num_processes == 2
+
+    class DummyIterableDataset(IterableDataset):
+        def __init__(self, data):
+            self.data = data
+
+        def __len__(self):
+            return len(self.data)
+
+        def __iter__(self):
+            for element in self.data:
+                yield element
+
+    iterable_dataset = DummyIterableDataset(torch.as_tensor(range(30)))
+    dataloader = DataLoader(iterable_dataset, batch_size=4)
+
+    accelerator = Accelerator()
+    prepared_dataloader = accelerator.prepare(dataloader)
+
+    assert type(prepared_dataloader) == DataLoaderDispatcher
+
+    if state.is_main_process:
+        logger = logging.root.manager.loggerDict["accelerate.accelerator"]
+        list_handler = ListHandler()
+        logger.addHandler(list_handler)
+
+    batches_for_metrics = []
+    for batch in prepared_dataloader:
+        batches_for_metrics.append(accelerator.gather_for_metrics(batch))
+
+    assert torch.cat(batches_for_metrics).size(0) == 30
+
+    if state.is_main_process:
+        assert len(list_handler.logs) == 0
+
+        logger.removeHandler(list_handler)
```

Comment:
This should be in `test_utils/scripts/external_metrics/test_metrics.py`

Reply:
Addressed in [e35136e](https://github.com/huggingface/accelerate/pull/1784/commits/e35136ea1f35ce5630ff470f39d23c720ff48111)